src/Uno.UWP/Devices/Geolocation/Geolocator.wasm.cs

```diff
@@ -0,0 +1,251 @@
+			}
+
+			return 0;
+		}
+
+		[Preserve]
+		public static int DispatchError(string currentPositionRequestResult, string requestId)
+		{
+			if (_pendingGeopositionRequests.TryRemove(requestId, out var geopositionCompletionSource))
+			{
+				if (!Enum.TryParse<PositionStatus>(currentPositionRequestResult, out var positionStatus))
+				{
+					throw new ArgumentOutOfRangeException(
+						nameof(currentPositionRequestResult),
+						"DispatchError argument must be a serialzied PositionStatus");
+				}
+				BroadcastStatus(positionStatus);
+				switch (positionStatus)
+				{
+					case PositionStatus.NoData:						
+						geopositionCompletionSource.SetException(
+							new Exception("This operation returned because the timeout period expired."));
+						break;
+					case PositionStatus.Disabled:
+						geopositionCompletionSource.SetException(
+							new UnauthorizedAccessException(
+								"Access is denied. Your App does not have permission to access location data. " +
+								"The user blocked access, or the device location services are currently turned off."));
+						break;
+					case PositionStatus.NotAvailable:
+						geopositionCompletionSource.SetException(
+							new InvalidOperationException("Location services API is not available."));
+						break;
+					default:
+						throw new ArgumentOutOfRangeException(
+							nameof(currentPositionRequestResult), "DispatchError position status must be an error status");
+				}
+			}
+			return 0;
+		}
+
+		/// <summary>
+		/// Handles result of Geolocation access request (initiated by <see cref="RequestAccessAsync"/>.
+		/// </summary>
+		/// <param name="serializedAccessStatus">Serialized string value from the <see cref="GeolocationAccessStatus"/> enum</param>
+		/// <returns>0 - needed to bind method from WASM</returns>
+		[Preserve]
+		public static int DispatchAccessRequest(string serializedAccessStatus)
+		{
+			if (serializedAccessStatus is null)
+			{
+				throw new ArgumentNullException(nameof(serializedAccessStatus));
+			}
+			if (!Enum.TryParse<GeolocationAccessStatus>(serializedAccessStatus, out var geolocationAccessStatus))
+			{
+				throw new ArgumentOutOfRangeException(
+					nameof(serializedAccessStatus),
+					"Parameter must be parseable as GeolocationAccessStatus enum value");
+			}
+
+			lock (_pendingAccessRequests)
+			{
+				//resolve all pending requests
+				foreach (var request in _pendingAccessRequests)
+				{
+					request.SetResult(geolocationAccessStatus);
+				}
+				_pendingAccessRequests.Clear();
+			}
+			return 0;
+		}
+
+		/// <summary>
+		/// Parses geocoordinate retrieved from JS in form of colon-separated string
+		/// </summary>
+		/// <param name="serializedGeoposition">Serialized, colon-separated position</param>
+		/// <returns>Geocoordinate</returns>
+		private static Geocoordinate ParseGeocoordinate(string serializedPosition)
+		{
+			var dataSplit = serializedPosition.Split(':');
+
+			var latitude = double.Parse(dataSplit[0]);
```

Comment:
Use invariant culture for the parsing, same for all parsing below

Reply:
Done